### PR TITLE
test: unify test config

### PR DESCRIPTION
**Pull request by ricokahler**
### Description

Unifies more of the testing config to make it easier to write tests! (Shoutout to @bjoerge for the initial work unifying this 🙌)

Specially, this PR includes a naive-but-still-useful parts aliasing mechanism.

### What to review

See the logic in the `jest.config.base.js`

### CODE REVIEW

File: .module-aliases.js [hunk 1]
```diff
@@ -3,10 +3,12 @@
 //
 // this file is currently read by:
 // - webpack when running the dev server
+// - vite config for the workshop
 // - jest when running test suite
-// note: order matters. The most specific paths should appear first
```
Comment (bjoerge) [inline .module-aliases.js:7]: order doesn't matter anymore?

File: jest.config.base.js [hunk 2]
```diff
@@ -1,51 +1,147 @@
 const path = require('path')
```
Comment (bjoerge) [inline jest.config.base.js:1]: maybe the file name of this one should reflect that it's no longer exporting a jest config, but a config creator? Rename to `createJestConfig.ts` or something?

File: jest-setup.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const path = require('path')
```
Comment (bjoerge) [inline jest-setup.js:1]: maybe this could live in the `test` directory as well?

**Review by bjoerge — APPROVED** [review]
LGTM! Left some minor comments inline.

File: .module-aliases.js (hunk 1 above)
Comment (ricokahler) [reply .module-aliases.js:7]: yeah i guess it doesn't. nice catch!

**Pull request merged**